NitroxPatcher/Patches/SubNameInput_OnColorChange_Patch.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Harmony;
+using NitroxClient.GameLogic.Helper;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using System;
+using System.Reflection;
+using NitroxClient;
+
+namespace NitroxPatcher.Patches
+{
+    public class SubNameInput_OnColorChange_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(SubNameInput);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("OnDeselect", BindingFlags.Public | BindingFlags.Instance);
+        public static readonly MethodInfo TARGET_METHOD2 = TARGET_CLASS.GetMethod("SetSelected", BindingFlags.Public | BindingFlags.Instance);
```

Comment:
This new system doesn't work nearly as well as in V1.3: Tab selection is still not synced, and color now only updates once you switch tabs.

I think you need to put the patch for `OnColorChange` back, and call:
```csharp
screenProxy.subNameInput.SetSelected(colorPacket.Index);
```
in `CyclopsChangeColorProcessor`.

Also: I checked, and [my commit](https://github.com/MarijnS95/Nitrox/commit/4e6b509479539b8e7c7a7fda85a90f1ea40b304a#diff-c87f3452addb749c011d5ed239b64330) for supplying patch-functions to `this.Patch*` is on an outdated branch, but I can put it on my current branch if that makes it easier for you to implement multiple related patches here.

Reply:
Ok the color Picker is not sync I will fix this. But if I sync the the tab with SetSelected() changes from an other person will be deleted if you change your tab...
In fact the patcher with OnColorChange() as trigger will produce much server traffic so I use OnDeselect() and SetSelected()...

Reply:
If the color itself is constantly in sync, then no changes should be lost. For now I think it's okay that players can use the console simultaneously and overwrite eachothers changes - in the future (once the headless server is done) I'll update and merge the ownership framework, so that we have a uniform way to disable access to something when another player is using it.

You can limit the amount of packets sent, as long as you make sure the final color is sent (and put this packet in the logging blacklist). As a bonus, you can interpolate the color over time so the transition looks smooth.